### PR TITLE
[docs][yugabyted][2024.1.2] Add Read Replica and xCluster examples

### PR DESCRIPTION
**Pull request by gargsans-yb**
Summary:
* Adding `xcluster` command and it's respective sub-commands.
* Adding examples for read-replica deployments
* Adding examples for setting-up xcluster replication.
* Adding examples on how to use CSV value master and tserver gflags.
* Updating example on Upgrading a YugabyteDB cluster to a newer version using `finalize_upgrade` command.

DOC-414
DOC-386

### CODE REVIEW

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 1]
```diff
@@ -1165,15 +1302,15 @@ To create a secure multi-zone cluster:
 
 To create a multi-zone cluster:
 
-1. Start the first node by running the `yugabyted start` command, passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details, as follows:
+1. Start the first node by running the `yugabyted start` command, passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1305]:
```suggestion
1. Start the first node by running the `yugabyted start` command, passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

     Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value.
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 2]
```diff
@@ -1165,15 +1302,15 @@ To create a secure multi-zone cluster:
...
 
     ```sh
     ./bin/yugabyted start --advertise_address=<host-ip> \
         --cloud_location=aws.us-east-1.us-east-1a \
         --fault_tolerance=zone
     ```
 
-1. Start the second and the third node on two separate VMs using the `--join` flag, as follows:
+1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1313]:
```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 3]
```diff
@@ -1229,7 +1366,7 @@ You can set the replication factor of the cluster manually using the `--rf` flag
 
 To create a secure multi-region cluster:
 
-1. Start the first node by running the `yugabyted start` command, using the `--secure` flag and passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details, as follows:
+1. Start the first node by running the `yugabyted start` command, using the `--secure` flag and passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1369]:
```suggestion
1. Start the first node by running the `yugabyted start` command, using the `--secure` flag and passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 4]
```diff
@@ -1248,7 +1385,7 @@ To create a secure multi-region cluster:
     - Copy the certificates for the second VM from `$HOME/var/generated_certs/<IP_of_VM_2>` in the first VM to `$HOME/var/certs` in the second VM.
     - Copy the certificates for third VM from `$HOME/var/generated_certs/<IP_of_VM_3>` in first VM to `$HOME/var/certs` in the third VM.
 
-1. Start the second and the third node on two separate VMs using the `--join` flag, as follows:
+1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1388]:
```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 5]
```diff
@@ -1270,15 +1407,15 @@ To create a secure multi-region cluster:
 
 To create a multi-region cluster:
 
-1. Start the first node by running the `yugabyted start` command, pass in the `--cloud_location` and `--fault_tolerance` flags to set the node location details as follows:
+1. Start the first node by running the `yugabyted start` command, pass in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1410]:
```suggestion
1. Start the first node by running the `yugabyted start` command, pass in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value.
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 6]
```diff
@@ -1270,15 +1407,15 @@ To create a secure multi-region cluster:
...
 
     ```sh
     ./bin/yugabyted start --advertise_address=<host-ip> \
         --cloud_location=aws.us-east-1.us-east-1a \
         --fault_tolerance=region
     ```
 
-1. Start the second and the third node on two separate VMs using the `--join` flag, as follows:
+1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yugabyted.md:1418]:
```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 7]
```diff
@@ -912,6 +915,140 @@ Usage: yugabyted version [flags]
...
+: Print the command-line help and exit.
+
+--base_dir *base-directory*
+: The base directory for the yugabyted server.
+
+--databases *xcluster-databases*
+: Comma separated list of databases to be added to the replication.
+
+--replication_id *xcluster-replication-id*
+: A unique string to be defined by you which will be used to uniquely identify the replication.
```
[the remark this answers is not in the document]
Comment (ddhodge) [reply docs/content/preview/reference/configuration/yugabyted.md:959]: User creates the replication ID using this command? Or enters an existing ID?

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 8]
```diff
@@ -685,7 +688,9 @@ Determine the status of a restore task:
 
 Use the `yugabyted start` command to start a one-node YugabyteDB cluster for running [YSQL](../../../api/ysql) and [YCQL](../../../api/ycql) workloads in your local environment.
 
-Note that to use encryption in transit, OpenSSL must be installed on the nodes.
+To use encryption in transit, OpenSSL must be installed on the nodes.
+
+If you want to use backup and restore, start the node with `--backup_daemon=true` to initialize the backup and restore agent. You also need to download and extract the [YB Controller release](https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz) to the yugabyte-{{< yb-version version="preview" >}} folder.
```
Comment (nchandrappa) [inline docs/content/preview/reference/configuration/yugabyted.md:693]: can we please update the download string to the following - 

```
https://downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz
```

Comment (nchandrappa) [reply docs/content/preview/reference/configuration/yugabyted.md:693]: Change from  - 

```
to the yugabyte-{{< yb-version version="preview" >}} folder.
```
to -

```
to the yugabyte-{{< yb-version version="preview" >}} release directory.

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 9]
```diff
@@ -764,7 +769,8 @@ For on-premises deployments, consider racks as zones to treat them as fault doma
 : Use this flag to start a read replica node.
 
 --backup_daemon *backup-daemon-process*
-: Enable or disable the backup daemon with yugabyted start. Default : `false`
+: Enable or disable the backup daemon with yugabyted start. Default: `false`
+: If you start a cluster using the `--backup_daemon` flag, you also need to download and extract the [YB Controller release](https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz) to the yugabyte-{{< yb-version version="preview" >}} folder.
```
Comment (nchandrappa) [inline docs/content/preview/reference/configuration/yugabyted.md:773]: Can we please update the download string to the following -

```
https://downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz
```

**Review by nchandrappa — APPROVED** [review]

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 10]
```diff
@@ -912,6 +915,140 @@ Usage: yugabyted version [flags]
...
+The following sub-commands are available for the `yugabyted xcluster` command:
+
+- [checkpoint](#checkpoint)
+- [set_up](#set-up)
+- [status](#status-1)
+- [delete](#delete-1)
+
+#### checkpoint
+
+Use the sub-command `yugabyted xcluster checkpoint` to checkpoint a new xcluster replication between 2 clusters. This command needs to be run from the source cluster of the replication.
```
[the remark this answers is not in the document]
Comment (nchandrappa) [reply docs/content/preview/reference/configuration/yugabyted.md]: This is a new xcluster feature in the coredb. Checkpoint will take a snapshot of the DB and determine whether bootstrapping is required for the remote cluster.

**Review by aishwarya24 — APPROVED** [review]
LGTM - Copy the changes to stable.

**Pull request merged**